exporter/collector/metricsexporter.go

```diff
@@ -79,6 +81,22 @@ func newGoogleCloudMetricsExporter(
 ) (component.MetricsExporter, error) {
 	setVersionInUserAgent(cfg, set.BuildInfo.Version)
 
+	// TODO - Share this lookup somewhere
+	if cfg.ProjectID == "" {
+		if ctx == nil {
```

Comment:
Checking for nil context is unusual in go.  Can we make sure the passed-in context is not nil everywhere and remove this check?

Reply:
this was copy-pasta from opencensus exporter, I can just remove.